examples/segmentation_camera/Main.cc

```diff
@@ -0,0 +1,214 @@
+  descriptor.mesh = meshManager->Load(descriptor.meshName);
+  MeshPtr meshGeom = _scene->CreateMesh(descriptor);
+  mesh->AddGeometry(meshGeom);
+  mesh->SetMaterial(yellow);
+  mesh->SetUserData("label", 5);
+  root->AddChild(mesh);
+
+  // create a box
+  VisualPtr box = _scene->CreateVisual("box");
+  box->SetLocalPosition(3, 1.5, 0);
+  box->SetLocalRotation(0, 0, 0.7);
+  GeometryPtr boxGeom = _scene->CreateBox();
+  box->AddGeometry(boxGeom);
+  box->SetMaterial(green);
+  box->SetUserData("label", 2);
+  root->AddChild(box);
+
+  // create a box
+  VisualPtr box2 = _scene->CreateVisual("box2");
+  box2->SetLocalPosition(2, -1, 1);
+  box2->SetLocalRotation(0, 0.3, 0.7);
+  GeometryPtr boxGeom2 = _scene->CreateBox();
+  box2->AddGeometry(boxGeom2);
+  box2->SetMaterial(green);
+  box2->SetUserData("label", 2);
+  root->AddChild(box2);
+
+  // create a sphere
+  VisualPtr sphere = _scene->CreateVisual("sphere");
+  sphere->SetLocalPosition(3, -1.5, 0);
+  GeometryPtr sphereGeom = _scene->CreateSphere();
+  sphere->AddGeometry(sphereGeom);
+  sphere->SetMaterial(red);
+  sphere->SetUserData("label", 3);
+  root->AddChild(sphere);
+
+  // create a sphere2
+  VisualPtr sphere2 = _scene->CreateVisual("sphere2");
+  sphere2->SetLocalPosition(5, 4, 2);
+  GeometryPtr sphereGeom2 = _scene->CreateSphere();
+  sphere2->AddGeometry(sphereGeom2);
+  sphere2->SetMaterial(red);
+  sphere2->SetUserData("label", 3);
+  root->AddChild(sphere2);
+
+  // create camera
+  SegmentationCameraPtr camera = _scene->CreateSegmentationCamera("camera");
+  camera->SetLocalPosition(0.0, 0.0, 0.5);
+  camera->SetLocalRotation(0.0, 0.0, 0.0);
+  camera->SetImageWidth(800);
+  camera->SetImageHeight(600);
+  camera->SetImageFormat(PixelFormat::PF_R8G8B8);
+  camera->SetFarClipPlane(10);
+  camera->SetNearClipPlane(1.0);
+  camera->SetAspectRatio(1.333);
+  camera->SetHFOV(IGN_PI / 2);
+  // camera->SetBackgroundColor(math::Color(255,255,255));
+  camera->EnableColoredMap(true);
+  // camera->SetSegmentationType(SegmentationType::Panoptic);
+  camera->SetSegmentationType(SegmentationType::Semantic);
+  root->AddChild(camera);
+}
+
+//////////////////////////////////////////////////
+CameraPtr createCamera(const std::string &_engineName)
+{
+  // create and populate scene
+  RenderEngine *engine = rendering::engine(_engineName);
+  if (!engine)
+  {
+    ignwarn << "Engine '" << _engineName
+              << "' is not supported" << std::endl;
+    return CameraPtr();
+  }
+  ScenePtr scene = engine->CreateScene("scene");
+  buildScene(scene);
+
+  // return camera sensor
+  SensorPtr sensor = scene->SensorByName("camera");
+  return std::dynamic_pointer_cast<Camera>(sensor);
+}
+
+//////////////////////////////////////////////////
+int main(int _argc, char** _argv)
+{
+  glutInit(&_argc, _argv);
+
+  common::Console::SetVerbosity(4);
+  std::vector<std::string> engineNames;
+  std::vector<CameraPtr> cameras;
+
+  engineNames.push_back("ogre2");
```

Comment:
can you include this logic?

https://github.com/WilliamLewww/ign-rendering/blob/cc618741a43d86eaab6d0419a205f5d862e336f2/examples/actor_animation/Main.cc#L180-L188